sympy/vector/coordsysrect.py

```diff
@@ -109,12 +156,11 @@ def __new__(cls, name, location=None, rotation_matrix=None,
         # they may actually be 'coincident' wrt the root system.
         if parent is not None:
             obj = super(CoordSys3D, cls).__new__(
-                cls, Symbol(name), location, parent_orient, parent)
+                cls, Symbol(name), transformation, parent)
```

Comment:
`transformation` has to be a SymPy type, if we are going to use a lambda, we need to use SymPy's `Lambda` (with capital L).